avocado/core/job.py

```diff
@@ -505,11 +506,18 @@ def _run(self):
             archive.create(filename, self.logdir)
         _TEST_LOGGER.info('Test results available in %s', self.logdir)
 
-        tests_status = not bool(failures)
-        if tests_status:
-            return exit_codes.AVOCADO_ALL_OK
-        else:
-            return exit_codes.AVOCADO_TESTS_FAIL
+        exit_status = exit_codes.AVOCADO_ALL_OK
+        for test_status, fail_class in summary.iteritems():
```

Comment:
So we only need the "summary" to get the return code, or do I miss something? I'm wondering if storing all these is necessary. The runner itself is driving the execution, so it could probably report `{PASS, FAIL, INTERRUPTED}`. Later when we add support for multiple statuses, it could report list of statuses and this could add JOB_ERROR in case the job error.